pkg/machines/scripts/create_machine.sh

```diff
@@ -63,6 +65,32 @@ else
     UNATTENDED_PARAMS=""
 fi
 
+if [ "$USE_CLOUD_INIT" = "true" ]; then
+    if [ -z "$USER_PASSWORD" ]; then
+        USER_LINE=""
+        CREATE_USER_LINE=""
+    else
+        USER_LINE="    $USER_LOGIN:$USER_PASSWORD"
+        CREATE_USER_LINE="users:"$'\n'"  - name: $USER_LOGIN"
```

Comment:
I'm actually surprised that this `'\n'` thingy works -- shell is usually rather picky/broken with nested spacing. But so much the better!

FTR, it'd be cleaner to build it piece-wise, like:
```sh
echo '#cloud-config' >> $USER_DATA_FILE
if [ -n "$VAR1" ]; then
    echo "    user=$VAR1" >> $USER_DATA_FILE
```
and so on. But not necessary to change it now.